### ISSUE TITLE
Deprecate Postgis in our database and API

### ISSUE THREAD

**Issue by znarf** [opencollective/opencollective#3171]
In our API and in our database schema, we depend on Postgis.

This is adding complexity in developer setup for not much use.

We should aim to stop depending on it.

**Comment by Betree** [comment]
As part of https://github.com/opencollective/opencollective-api/pull/7457, we're moving the only column that depends on Postgis to `JSONB` thus removing the need for this dependency.

We'll need to have a second pass to update the docs and consolidate the migrations: if we just ask people to use postgres without `postgis`, their first `npm run db:migrate` will fail when trying to migrate the column.

To summarize, the next steps are:
- [ ] Update `test/dbdumps/opencollective_dvl.pgsql` to make sure it includes all the current migrations
- [ ] Move all migrations to `migrations/archives`
- [ ] Remove all references to postgis from the code, CI config files and documentation

**Comment by Betree** [comment]
Depends on https://github.com/opencollective/opencollective/issues/5515

**Comment by Betree** [comment]
Postgis is gone! I'll keep an eye on our channels to look for regressions
